websites/C/Cisco Netwoking Academy/presence.ts

```diff
@@ -0,0 +1,55 @@
+const presence = new Presence({
+  clientId: "858886158045806602"
+});
+
+let prevURL = "",
+  timestamp = Date.now(),
+  details = "",
+  state = "";
+presence.on("UpdateData", async () => {
+  if (prevURL !== window.location.pathname) {
+    prevURL = window.location.pathname;
+    timestamp = Date.now();
+  }
+  if (window.location.pathname === "/") 
+    details = "Browsing Home Page";
+  else if (window.location.pathname === "/portal/learning") 
+    details = "Netacad Portal";
+  else if (window.location.pathname.startsWith("/course/")) {
+    details = "Viewing course";
+    state = document.getElementsByTagName("h3")[0].innerText;
+  } else if (window.location.pathname.startsWith("/grade/report/")) {
+    details = "Viewing grades";
+    state = document.getElementsByTagName("h3")[0].innerText;
+  } else if (window.location.pathname.startsWith("/local/mail/")) 
+    details = "Viewing messages";
+  else if (window.location.pathname.startsWith("/calendar/")) {
+    details = "Viewing calendar";
+    state = document.getElementsByTagName("h3")[0].innerText;
+  } else if (window.location.pathname.startsWith("/mod/")) {
+    if (document.getElementsByTagName("h2")[0].innerText.toUpperCase().includes("EXAM")) 
+      details = "Viewing exam";
+    else 
+      details = "Viewing course content";
+  
+    state = document.getElementsByTagName("h3")[0].innerText;
+  } else if (window.location.pathname.startsWith("/srwe-dl/")) {
+    details = "Viewing course content";
+    state = document.getElementsByTagName("h1")[0].innerText;
+  } else if (window.location.pathname.includes("assessment_history")) 
+    details = "Viewing Assesment History";
+  else 
+    details = "Browsing";
+  
+
+  const presenceData: PresenceData = {
+    largeImageKey: "netacadlogo",
+    details,
+    state,
```

Comment:
details and state could be "" right now, make sure not to set it if it is, or use
delete presenceData.details
to remove it again.

Reply:
I added an if statement below the presenceData part to delete state if it doesn't exist.
Details doesn't need to be deleted since it will always be set.